src/pokeService.js

```diff
@@ -17,6 +24,51 @@ async function getPokemonMoves(id) {
   return resolvePromises;
 }
 
+async function getPokemonEvoChain(id) {
+  const species = await getPokemonSpecies(id);
+  const evoChainUrl = species.data.evolution_chain.url;
+  const evoChains = await axios.get(evoChainUrl);
+  let evoData = evoChains.data.chain;
+
+  const evoChain = [];
+  do {
+    const numOfEvolutions = evoData.evolves_to.length;
+    let pokemon = await getPokemonByName(evoData.species.name);
+    evoChain.push({
+      url: evoData.species.url,
+      name: evoData.species.name,
+      id: pokemon.id,
+      image: pokemon.sprites.front_default,
+      min_level: !evoData.evolves_to[0] ? 1 : evoData.evolves_to[0].evolution_details[0].min_level,
+
+    });
+
+    if (numOfEvolutions > 1) {
+      for (let i = 1; i < numOfEvolutions; i++) {
+        pokemon = await getPokemonByName(evoData.evolves_to[i].species.name);
+        evoChain.push({
```

Comment:
You could probably use a `.map` on the result of your Promise.all here.
Something along the lines of:
```javascript
// I am winging this code don't expect it to work
Promise.all(evoData.evolved_to.map(pokemon => getPokemonByName(pokemon.species.name))
.map(pokemon => ({
  url: pokemon....
  name: ...
})
```